backend/src/load.rs

```diff
@@ -0,0 +1,418 @@
+                let components =
+                    &filename[(filename.find("--").unwrap() + 2)..filename.find(".json").unwrap()]
+                    .split('-')
+                    .map(|x| x.parse().unwrap())
+                    .collect::<Vec<u64>>();
+
+                // TODO: Use a ymd constructor and then use and_time to save
+                // the round-trip through string. Alternatively, remove the
+                // calls to parse above.
+                NaiveDateTime::parse_from_str(
+                    &format!("{}-{}-{} {}:{}:{}",
+                        components[0],
+                        components[1],
+                        components[2],
+                        components[3],
+                        components[4],
+                        components.get(5).unwrap_or(&0)), "%Y-%m-%d %H:%M:%S")
+                    .map_err(|err| err.into())
+            }
+
+            let header = &contents["header"];
+            let date = header["date"].as_str().unwrap();
+            let date = try!(parse_from_header(&date).or_else(|_| parse_from_filename(&filename)));
+
+            let test_name = &filename[0..filename.find("--").unwrap()];
+
+            let times = &contents["times"];
+            if test_name == "rustc" {
+                c_bootstraps += 1;
+                data_rustc.push(Data::new(date, &header, times, true));
+
+                for timing in times.members() {
+                    let crate_name = timing["crate"].as_str().unwrap().to_string();
+                    crate_list.insert(crate_name);
+                }
+            } else {
+                benchmarks.insert(test_name.to_string());
+                let index = data_benchmarks.iter().position(|benchmark: &Data| benchmark.date == date);
+                if let Some(index) = index {
+                    c_benchmarks_add += 1;
+                    let crate_name = times[0]["crate"].as_str().unwrap();
+                    data_benchmarks[index].by_crate.insert(test_name.to_string(),
+                        make_times(times, false).remove(crate_name).unwrap());
+                } else {
+                    c_benchmarks += 1;
+                    data_benchmarks.push(Data::new(date, &header, times, false));
+                }
+            }
+
+            for timing in times.members() {
+                for (phase, _) in timing["times"].entries() {
+                    phase_list.insert(phase.to_string());
+                }
+            }
+
+            if last_date.is_none() || last_date.as_ref().unwrap() < &date {
+                last_date = Some(date);
+            }
+        }
+
+        let last_date = last_date.expect("No dates found");
+
+        data_rustc.sort();
+        data_benchmarks.sort();
+
+        // Post processing to generate the summary data.
+        let summary_rustc = Summary::new(&data_rustc, last_date);
+        let summary_benchmarks = Summary::new(&data_benchmarks, last_date);
```

Comment:
I just did a test, and both of these complete in less than a second (in debug mode). Should we move them into the server logic, rather than splitting the summary into two steps? I don't see many advantages to doing so, but wanted to leave a comment noting that here.